planetaryimage/image.py

```diff
@@ -71,184 +61,76 @@ def __init__(self, stream, filename=None, compression=None, memory_layout='DISK'
-        self.data = self._parse_data(stream)
+        self.data = self._load_data(stream)
 
     def __repr__(self):
+        # TODO: pick a better repr
         return self.filename
 
-    @staticmethod
-    def get_nested_dict(item, key_list):
-        """Get value from nested dictionary using a key list. """
-        return reduce(lambda x, y: x[y], key_list, item)
-
-    def apply_scaling(self, copy=True):
-        """Scale pixel values to there true DN.
-
-        Parameters
-        ----------
-        copy : boolean
-            whether to apply the scalling to a copy of the pixel data
-            and leave the orginial unaffected
-
-        Returns
-        -------
-        scaled_pixel_data
-            a scaled version of the pixel data
-        """
-        if copy:
-            return self.multiplier * self.data + self.base
-
-        if self.multiplier != 1:
-            self.data *= self.multiplier
-
-        if self.base != 0:
-            self.data += self.base
-
-        return self.data
-
-    def apply_numpy_specials(self, copy=True):
-        """Convert isis special pixel values to numpy special pixel values.
-
-            =======  =======
-             Isis     Numpy
-            =======  =======
-            Null     nan
-            Lrs      -inf
-            Lis      -inf
-            His      inf
-            Hrs      inf
-            =======  =======
-
-        Parameters
-        ----------
-        copy : boolean
-            whether to apply the new special values to a copy of the
-            pixel data and leave the orginial unaffected
-
-        Returns
-        -------
-        modified_image_data
-            a numpy array with special values converted to numpy's ``nan``,
-            ``inf`` and ``-inf``
-        """
-        if copy:
-            data = self.data.astype(numpy.float64)
-
-        elif self.data.dtype != numpy.float64:
-            data = self.data = self.data.astype(numpy.float64)
-
-        else:
-            data = self.data
-
-        data[data == self.specials['Null']] = numpy.nan
-        data[data < self.specials['Min']] = numpy.NINF
-        data[data > self.specials['Max']] = numpy.inf
-
-        return data
-
-    def specials_mask(self):
-        """Create a pixel map for special pixels.
-
-        Returns
-        -------
-        mask_array
-            an array where the value is `False` if the pixel is special
-            and `True` otherwise
-        """
-        mask = self.data >= self.specials['Min']
-        mask &= self.data <= self.specials['Max']
-        return mask
-
-    def get_image_array(self):
-        """Create an array for use in making an image.
-
-        Creates a linear stretch of the image and scales it to between `0` and
-        `255`. `Null`, `Lis` and `Lrs` pixels are set to `0`. `His` and `Hrs`
-        pixels are set to `255`.
-
-        Usage::
-
-            from pysis import CubeFile
-            from PIL import Image
+    @property
```

Comment:
I have added this `image` property for the time being so we have a usable interface.